sdk/maintenance/Azure.ResourceManager.Maintenance/src/Custom/Models/MaintenanceConfigurationData.Serialization.cs

```diff
@@ -5,100 +5,26 @@
-            {
-                writer.WritePropertyName("namespace"u8);
-                writer.WriteStringValue(Namespace);
-            }
-            if (Optional.IsCollectionDefined(ExtensionProperties))
-            {
-                writer.WritePropertyName("extensionProperties"u8);
-                writer.WriteStartObject();
-                foreach (var item in ExtensionProperties)
-                {
-                    writer.WritePropertyName(item.Key);
-                    writer.WriteStringValue(item.Value);
-                }
-                writer.WriteEndObject();
-            }
-            if (Optional.IsDefined(MaintenanceScope))
-            {
-                writer.WritePropertyName("maintenanceScope"u8);
-                writer.WriteStringValue(MaintenanceScope.Value.ToString());
-            }
-            if (Optional.IsDefined(Visibility))
-            {
-                writer.WritePropertyName("visibility"u8);
-                writer.WriteStringValue(Visibility.Value.ToString());
-            }
-            if (Optional.IsDefined(InstallPatches))
-            {
-                writer.WritePropertyName("installPatches"u8);
-                writer.WriteObjectValue(InstallPatches);
-            }
-            writer.WritePropertyName("maintenanceWindow"u8);
-            writer.WriteStartObject();
-            // Custom code to support custom datetime format for the 2 properties startDateTime and expirationDateTime
-            // Here is an issue https://github.com/Azure/autorest.csharp/issues/3184 to track if codegen can support custom datetime format in the furture.
-            if (Optional.IsDefined(StartOn))
-            {
-                writer.WritePropertyName("startDateTime"u8);
-                writer.WriteStringValue(StartOn.Value, "yyyy-MM-dd hh:mm");
-            }
-            // Custom code to support custom datetime format for the 2 properties startDateTime and expirationDateTime
-            // Here is an issue https://github.com/Azure/autorest.csharp/issues/3184 to track if codegen can support custom datetime format in the furture.
-            if (Optional.IsDefined(ExpireOn))
-            {
-                writer.WritePropertyName("expirationDateTime"u8);
-                writer.WriteStringValue(ExpireOn.Value, "yyyy-MM-dd hh:mm");
-            }
-            if (Optional.IsDefined(Duration))
-            {
-                writer.WritePropertyName("duration"u8);
-                writer.WriteStringValue(Duration.Value, "c");
-            }
-            if (Optional.IsDefined(TimeZone))
-            {
-                writer.WritePropertyName("timeZone"u8);
-                writer.WriteStringValue(TimeZone);
-            }
-            if (Optional.IsDefined(RecurEvery))
-            {
-                writer.WritePropertyName("recurEvery"u8);
-                writer.WriteStringValue(RecurEvery);
-            }
+            writer.WriteStringValue(StartOn.Value, "yyyy-MM-dd HH:mm");
+        }
```

Comment:
actually we could put these two methods in the same partial class as those properties are which reduce the amount of file and makes it easier to follow

Reply:
You mean put them in custom `MaintenanceConfigurationData.cs` instead of `MaintenanceConfigurationData.Serialization.cs`? They have different purpose though, thought serialization related logic goes to `MaintenanceConfigurationData.Serialization.cs`

Reply:
Like the existing one in [ImportJob.Serialziation.cs](https://github.com/Azure/azure-sdk-for-net/blob/8fa7955ac9cd1cf2c124f7d387ef07c2c5a53387/sdk/digitaltwins/Azure.DigitalTwins.Core/src/Customized/Models/ImportJob.Serialization.cs#L106), prefer to keep it in `MaintenanceConfigurationData.Serialization.cs`